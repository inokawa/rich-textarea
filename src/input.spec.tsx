import { it, describe, expect, afterEach } from "vitest";
import { cleanup, render } from "@testing-library/react";
import { experimental_RichInput as RichInput } from ".";

global.ResizeObserver = class mockResizeObjerver {
  instanceResize: ResizeObserver | null = null;
  callbackResize: ResizeObserverCallback | null = null;
  constructor(callback: ResizeObserverCallback) {
    this.instanceResize = this;
    this.callbackResize = callback;
  }
  disconnect() {}
  observe(_target: Element, _options?: ResizeObserverOptions) {}
  unobserve(_target: Element) {}
};

const NOP = () => {};

afterEach(cleanup);

describe("smoke", () => {
  it("controlled", () => {
    const { asFragment } = render(
      <RichInput value={"Lorem ipsum dolor sit amet"} onChange={NOP}>
        {(v) => {
          return v.split("").map((t, i) => (
            <span key={i} style={{ color: i % 2 === 0 ? "red" : undefined }}>
              {t}
            </span>
          ));
        }}
      </RichInput>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("uncontrolled", () => {
    const { asFragment } = render(
      <RichInput defaultValue={"Lorem ipsum dolor sit amet"}>
        {(v) => {
          return v.split("").map((t, i) => (
            <span key={i} style={{ color: i % 2 === 0 ? "red" : undefined }}>
              {t}
            </span>
          ));
        }}
      </RichInput>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
