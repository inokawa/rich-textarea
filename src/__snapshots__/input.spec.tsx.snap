// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke controlled 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; position: relative; width: 0px; height: 0px;"
  >
    <div
      style="position: absolute; overflow: hidden; top: 0px; left: 0px; width: 0px; height: 0px;"
    >
      <div
        aria-hidden="true"
        style="padding: 1px 1px 1px 1px; margin: 0px 0px 0px 0px; border: 2px inset; border-width: 2px; border-top-width: 2px; border-bottom-width: 2px; border-left-width: 2px; border-right-width: 2px; border-style: inset; border-top-style: inset; border-bottom-style: inset; border-left-style: inset; border-right-style: inset; font-family: -webkit-small-control; text-align: start; text-transform: none; text-indent: 0; letter-spacing: normal; word-spacing: normal; line-height: normal; border-color: transparent;"
      >
        <div
          style="height: 0px;"
        >
          <div
            style="transform: translate(0px, 0px); pointer-events: none; user-select: none; box-sizing: content-box; white-space: pre;"
          >
            <span
              style="color: red;"
            >
              L
            </span>
            <span>
              o
            </span>
            <span
              style="color: red;"
            >
              r
            </span>
            <span>
              e
            </span>
            <span
              style="color: red;"
            >
              m
            </span>
            <span>
               
            </span>
            <span
              style="color: red;"
            >
              i
            </span>
            <span>
              p
            </span>
            <span
              style="color: red;"
            >
              s
            </span>
            <span>
              u
            </span>
            <span
              style="color: red;"
            >
              m
            </span>
            <span>
               
            </span>
            <span
              style="color: red;"
            >
              d
            </span>
            <span>
              o
            </span>
            <span
              style="color: red;"
            >
              l
            </span>
            <span>
              o
            </span>
            <span
              style="color: red;"
            >
              r
            </span>
            <span>
               
            </span>
            <span
              style="color: red;"
            >
              s
            </span>
            <span>
              i
            </span>
            <span
              style="color: red;"
            >
              t
            </span>
            <span>
               
            </span>
            <span
              style="color: red;"
            >
              a
            </span>
            <span>
              m
            </span>
            <span
              style="color: red;"
            >
              e
            </span>
            <span>
              t
            </span>
            <span
              style="color: transparent;"
            >
              ​
            </span>
          </div>
        </div>
      </div>
    </div>
    <input
      style="background: transparent; margin: 0px; vertical-align: top; color: transparent;"
      value="Lorem ipsum dolor sit amet"
    />
  </div>
</DocumentFragment>
`;

exports[`smoke uncontrolled 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; position: relative; width: 0px; height: 0px;"
  >
    <div
      style="position: absolute; overflow: hidden; top: 0px; left: 0px; width: 0px; height: 0px;"
    >
      <div
        aria-hidden="true"
        style="padding: 1px 1px 1px 1px; margin: 0px 0px 0px 0px; border: 2px inset; border-width: 2px; border-top-width: 2px; border-bottom-width: 2px; border-left-width: 2px; border-right-width: 2px; border-style: inset; border-top-style: inset; border-bottom-style: inset; border-left-style: inset; border-right-style: inset; font-family: -webkit-small-control; text-align: start; text-transform: none; text-indent: 0; letter-spacing: normal; word-spacing: normal; line-height: normal; border-color: transparent;"
      >
        <div
          style="height: 0px;"
        >
          <div
            style="transform: translate(0px, 0px); pointer-events: none; user-select: none; box-sizing: content-box; white-space: pre;"
          >
            <span
              style="color: red;"
            >
              L
            </span>
            <span>
              o
            </span>
            <span
              style="color: red;"
            >
              r
            </span>
            <span>
              e
            </span>
            <span
              style="color: red;"
            >
              m
            </span>
            <span>
               
            </span>
            <span
              style="color: red;"
            >
              i
            </span>
            <span>
              p
            </span>
            <span
              style="color: red;"
            >
              s
            </span>
            <span>
              u
            </span>
            <span
              style="color: red;"
            >
              m
            </span>
            <span>
               
            </span>
            <span
              style="color: red;"
            >
              d
            </span>
            <span>
              o
            </span>
            <span
              style="color: red;"
            >
              l
            </span>
            <span>
              o
            </span>
            <span
              style="color: red;"
            >
              r
            </span>
            <span>
               
            </span>
            <span
              style="color: red;"
            >
              s
            </span>
            <span>
              i
            </span>
            <span
              style="color: red;"
            >
              t
            </span>
            <span>
               
            </span>
            <span
              style="color: red;"
            >
              a
            </span>
            <span>
              m
            </span>
            <span
              style="color: red;"
            >
              e
            </span>
            <span>
              t
            </span>
            <span
              style="color: transparent;"
            >
              ​
            </span>
          </div>
        </div>
      </div>
    </div>
    <input
      style="background: transparent; margin: 0px; vertical-align: top; color: transparent;"
      value="Lorem ipsum dolor sit amet"
    />
  </div>
</DocumentFragment>
`;
